### DISCUSSION TITLE
[Share] Collapse all libraries

### DISCUSSION THREAD

**Discussion by cboulanger** [windingwind/zotero-actions-tags#406] [Action Scripts]
### Description

When you have many shared libraries open with their (sub)collection expanded, it is sometimes hard to access a particular library without scrolling for a long time. This action collapses all libraries so you can click on the one you need immediately. 

### Event

None

### Operation

Script

### Data

```javascript
/**
 * A description of this script.
 * @author Zotero Community
 * @usage 
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/
 * @see https://github.com/windingwind/zotero-actions-tags/discussions/
 */
if(item) return;
ZoteroPane = Zotero.getActiveZoteroPane();
Zotero.Libraries.getAll().forEach(lib => 
    ZoteroPane.collectionsView.collapseLibrary(lib.libraryID))
```

**Comment by windingwind** [comment]
Don't forget to add a line `if(item) return`, so that it is only executed once (see readme)

**Reply by cboulanger** [reply]
Thanks, I'll add it.

**Reply by cboulanger** [reply]
There needs to be a companion script which does that recursively with collections (see https://github.com/windingwind/zotero-actions-tags/discussions/357) - I wonder if you happen to know the API methods to do that - cannot be very hard.

**Comment by datcu-octavian** [comment]
Is there any way to make this script toggle between collapsed/uncollapsed, triggered by a keyboard shortcut?
